src/Core/App.xaml.cs

```diff
@@ -331,18 +331,27 @@ void NavigateToSetPasswordPageAction()
                     || message.Command == "unlocked"
                     || message.Command == AccountsManagerMessageCommands.ACCOUNT_SWITCH_COMPLETED)
                 {
-                    if (message.Command == AccountsManagerMessageCommands.ACCOUNT_SWITCH_COMPLETED)
+                    var userVerificationMediatorService = ServiceContainer.Resolve<IFido2MakeCredentialConfirmationUserInterface>();
+                    if (message.Command == AccountsManagerMessageCommands.ACCOUNT_SWITCH_COMPLETED && userVerificationMediatorService.IsConfirmingNewCredential)
                     {
-                        var userVerificationMediatorService = ServiceContainer.Resolve<IFido2MakeCredentialConfirmationUserInterface>();
                         userVerificationMediatorService?.OnConfirmationException(new AccountSwitchedException());
                     }
-                    
+
                     lock (_processingLoginRequestLock)
                     {
                         // lock doesn't allow for async execution
                         CheckPasswordlessLoginRequestsAsync().Wait();
                     }
                 }
+                else if (message.Command == "navigateTo" && message.Data is NavigationTarget navigationTarget)
+                {
+                    ArgumentNullException.ThrowIfNull(MainPage);
```

Comment:
🤔 If this is plausible to happen in some weird known scenario, is the information enough on the exception to know that it happened in this message command? If not, we could add a bit more specifics in the exception message.

Reply:
Really not needed. It it happens it's already outside of the message command context.